sdmetrics/reports/multi_table/base_multi_table_report.py

```diff
@@ -60,14 +60,27 @@ def _validate_relationships(self, real_data, synthetic_data, metadata):
 
     def _validate_metadata_matches_data(self, real_data, synthetic_data, metadata):
         """Validate that the metadata matches the data."""
-        self.table_names = list(metadata['tables'].keys())
         for table in self.table_names:
             super()._validate_metadata_matches_data(
                 real_data[table], synthetic_data[table], metadata['tables'][table]
             )
 
         self._validate_relationships(real_data, synthetic_data, metadata)
 
+    def _validate(self, real_data, synthetic_data, metadata):
+        """Validate the inputs.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data.
+            synthetic_data (pandas.DataFrame):
+                The synthetic data.
+            metadata (dict):
+                The metadata of the table.
+        """
+        self.table_names = list(metadata['tables'].keys())
+        super()._validate(real_data, synthetic_data, metadata)
```

Comment:
minor: I think it makes more sense to store this in the `generate` method. It's not really related to validating and it seems like generate is being used to get all the info we need for the report, almost acting like an init

Reply:
I agree, done in [f150da4](https://github.com/sdv-dev/SDMetrics/pull/524/commits/f150da4401f7f2a3eaa684a234036ef121ed7ae5)